### ISSUE TITLE
Imgbot not working

### ISSUE THREAD

**Issue by JouriR** [dabutvin/Imgbot#757]
So I've installed the img bot on my account a day ago but I still haven't received any pull requests. Did I do something wrong?

**Comment by Bonepoppa1** [comment]
757

**Comment by Bonepoppa1** [comment]
#759

**Comment by AlexEckstein** [comment]
I am having the same issue with a private repository that I have. Even tried adding the json config file to specify it to scan daily. No PR after about a month.

**Comment by hniV** [comment]
I'm having the same issue, only the first PR worked, after that, no PR appeared in my public repo.

**Comment by atin** [comment]
Having the same issue. By the way what is the average time to get a PR after optimization request? Is there a way to check the PR status?

**Comment by theodorusclarence** [comment]
Having the same issue, anybody has a fix on this?

**Comment by s-gbz** [comment]
Same here.

**Comment by zhaoweiguo** [comment]
mark, same issue

**Comment by s-gbz** [comment]
I encountered the same issue when I first installed Imgbot in my repo. It took about an hour for the initial PR to arrive.

**Comment by rusty1996** [comment]
Same here, it has been over two hours since installing Imgbot in my repository, and still no new pull requests. My repository has plenty of images that should have been noticed by now.

**Comment by kcak11** [comment]
Same issue here. ImgBot seems to work only for the first time and after that no further PRs are created when new images are added or edited. 
In certain cases, even the first PR doesn't come.
I am not sure if this App is being actively maintained or discontinued . . .

**Comment by norosa** [comment]
@GrigoreMihai imgbot is installed on https://github.com/doitintl/docs-gitbook-cmp and so far we haven't received any PRs at all (despite me manually requesting new optimizations)

**Comment by GrigoreMihai** [comment]
> @GrigoreMihai imgbot is installed on https://github.com/doitintl/docs-gitbook-cmp and so far we haven't received any PRs at all (despite me manually requesting new optimizations)

Hello, I checked the mentioned repository but I see only one request for optimization started, could you try again to request a new optimization again and let me know with a few minutes before you request the optimization so I can check some logs. ( since your repository is public I can fork it myself and test the optimization on it, but it will take a bit until we get the time to check it )

**Comment by BUCTPJP** [comment]
Same issue here. ImgBot seems to work only for the first time and after that no further PRs are created when new images are added or edited.

**Comment by GrigoreMihai** [comment]
> Same issue here. ImgBot seems to work only for the first time and after that no further PRs are created when new images are added or edited.

We checked your repository and after merging the first pull request the new added images where not meeting the optimization threshold, you can use the `minKBReduced`  and set it to 0 (the default behavior is not to spam a repository for every small image added) . You can find an example here: https://github.com/GrigoreMihai/MyWeb/blob/master/.imgbotconfig (here you can also see how the pr was opened after adding images and also automatically edited with the new images) .

**Comment by RodrigoGiraoSerrao** [comment]
> We checked your repository [...]

Sorry for insisting, @GrigoreMihai, but do you have any clue what is the issue that I am facing? Thanks 🙏

**Comment by GrigoreMihai** [comment]
> > We checked your repository [...]
> 
> Sorry for insisting, @GrigoreMihai, but do you have any clue what is the issue that I am facing? Thanks 🙏

Hello, 

Sure, we reviewed it from when you posted it, being a bigger repository it might not had the needed memory to be cloned for optimization, this issue is now fixed in the latest release, give it two retries (at a few hours apart) and let us know how it works.

**Comment by rodrigogiraoserrao** [comment]
> Sure, we reviewed it from when you posted it, being a bigger repository it might not had the needed memory to be cloned for optimization, this issue is now fixed in the latest release, give it two retries (at a few hours apart) and let us know how it works.

Thanks for getting back to me! I requested for an optimisation ~ 24 hours ago, and another one ~ 4 hours ago, and I haven't gotten a PR from the bot yet. Let me know if there is anything you'd like me to try. (Do I have to do something to update ImgBot..? Because I didn't do anything like that...)

**Comment by BUCTPJP** [comment]
Thank you for replying to me，and now I understand what's going on.In fact,after I replyed on github,I received&nbsp; pull request for some images.
------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "imgbot/Imgbot"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年2月25日(星期五) 凌晨0:52
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [imgbot/Imgbot] Imgbot not working (#757)





  
Sure, we reviewed it from when you posted it, being a bigger repository it might not had the needed memory to be cloned for optimization, this issue is now fixed in the latest release, give it two retries (at a few hours apart) and let us know how it works.
  
Thanks for getting back to me! I requested for an optimisation ~ 24 hours ago, and another one ~ 4 hours ago, and I haven't gotten a PR from the bot yet. Let me know if there is anything you'd like me to try. (Do I have to do something to update ImgBot..? Because I didn't do anything like that...)

**Comment by AnjaniGourisaria** [comment]
I just a ".imgbotconfig" file in my root directory of the repo to get the all daily but after day ther is no pull request from there side means imgbot
file content .imgbotconfig:

{
    "schedule": "daily",
    "aggressiveCompression": "true",
    "compressWiki": "true"
}

Kindly give the solutions for the problem

**Comment by GrigoreMihai** [comment]
> I just a ".imgbotconfig" file in my root directory of the repo to get the all daily but after day ther is no pull request from there side means imgbot file content .imgbotconfig:
> 
> { "schedule": "daily", "aggressiveCompression": "true", "compressWiki": "true" }
> 
> Kindly give the solutions for the problem

Hello, sorry for the late reply, can you please share the name of the repository so we can look into it ?

 (Also note that you need to have new commits with images to the repository for an optimization to trigger, the `daily` option just limits our optimization to once a day, also you could have not added enough images to have the minimum optimization threshold and for that you can set `minKBReduced`  to 0  and request a new optimization manually from https://imgbot.net/ )

**Comment by AnjaniGourisaria** [comment]
Hi Greeting ,

No Problem! 
Sure that i will note it 

Repo is https://github.com/AnjaniGourisaria/LeeTechnologies (Private)

Kindly give the solutions for the problem

**Comment by AnjaniGourisaria** [comment]
HI, Sorry for late reply ,can you please tell me how to send optimization request for this repository

I just visited to this link but it show my profile but not the request button

**Comment by GrigoreMihai** [comment]
> HI, Sorry for late reply ,can you please tell me how to send optimization request for this repository
> 
> I just visited to this link but it show my profile but not the request button

After you sign in into the page: https://imgbot.net/app/   you will see a list with all the repositories added to your account like this https://vertis.d.pr/N60uEQ .

 If you are not seeing any repository there we need to make sure they are added from the app settings https://github.com/settings/installations . Also please share a print of https://imgbot.net/app/ after you sign in if you are not seeing any repositories .

**Comment by AnjaniGourisaria** [comment]
Hi, Greeting ,
After Seeing that some project have "Request New Optimization" In near by snapshot there is some examples
Note: My repo is private
 
 ![image](https://user-images.githubusercontent.com/82677043/170467328-d259ab37-918c-488f-a7fa-aba468f2553b.png)

**Comment by GrigoreMihai** [comment]
> Hi, Greeting , After Seeing that some project have "Request New Optimization" In near by snapshot there is some examples Note: My repo is private
> 
> ![image](https://user-images.githubusercontent.com/82677043/170467328-d259ab37-918c-488f-a7fa-aba468f2553b.png)

You should also see private repositories in that list, if you are not seeing the repository it means it is not added in the app settings listed repositories here: https://github.com/settings/installations  select our app and click on configure https://vertis.d.pr/Kb4fXF . After that add the specific repository or select "All repositories" and click on save https://vertis.d.pr/C7alt2 .

**Comment by AnjaniGourisaria** [comment]
As You can see in  a given image full permission to the imgbot but still it give me no request in my private repo  but request new optimization for public repo only and i have many photo in that repo

![image](https://user-images.githubusercontent.com/82677043/170488508-0d3929e6-54bf-434f-a9f2-76de24e87aac.png)

**Comment by GrigoreMihai** [comment]
@AnjaniGourisaria I checked your account and you are on the open source plan, that is the reason why you cannot request manual optimization for that repository as we only allow public repositories on the open source plan. If you wish to also optimize private repositories consider upgrading your plan.

**Comment by AnjaniGourisaria** [comment]
Hi , Greeting ,
Firstly,
Thanks for your support 

I may think about to upgrade the plan for now no plans to upgrade ...

**Comment by WesBrownVShred** [comment]
Hi I am trialing a paid plan, agency, and haven't seen any PR's come through on the two private repos I have enabled it on. I can find nothing to help troubleshoot this except this issue.

**Comment by sobolevbel** [comment]
Hi, dear community. I have a similar problem. I've installed imgbot for my public repo using free plan, but still haven't received any PRs. Of course I granted required permissions and I see the repo at the imgbot app page. Could you please help me with it?
https://github.com/sobolevbel/jdg

**Comment by saipranay47** [comment]
hey @GrigoreMihai i am facing the similar problem with this repo - https://github.com/saipranay47/Aatmann

**Comment by A-JiuA** [comment]
I have the same issue. I have installed Imgbot for about a day, but haven't received PRs yet. I also request optimization from https://imgbot.net/app/ several times, the issue persists. Here is my repo: https://github.com/A-JiuA/Hexo_source
@GrigoreMihai

**Comment by hardingadonis** [comment]
I have the same issue. I have installed Imgbot, but haven't received PRs yet. I also try to request optimization manually from https://imgbot.net/app/.
My repo: https://github.com/hardingadonis/miu-shop
I'm working on `dev` branch.
@GrigoreMihai can u help me?

**Comment by Kushagra0347** [comment]
Hey @GrigoreMihai. I installed imgbot in my [repo](https://github.com/Kushagra0347/Portfolio_MYRON_EVANS)
. I've added .imgbotconfig as well in my frontend directory but still I haven't received any PRs and at the website it shows this 
![image](https://github.com/imgbot/Imgbot/assets/71805375/9e859f6c-6ad9-40e2-abe1-2265f8496e3f)

I want to know how much time it takes to get the first PR for a project of this size (Animation folder in public dir is 336mb for  context). Or is it a bug?

**Comment by JosefHobler** [comment]
I’m facing the same issue – I can’t access get pull request for my private repo while on the student plan. It’s been a month, and I’ve been manually trying to request it almost every day, but still no luck. Everything seems to be set up correctly on my end.

The only potential cause I can think of is that I recently changed my GitHub username.

**Comment by Andrewshin-7th-technology-student** [comment]
Hi, im having issues with Imgbot. It wont open any pr and my site is getting too heavy for the images that support it. Its been about 5 weeks and Imgbot is still not working. I've tried to request optimization for between hours, but it is not working. I added a `.imgbotconfig` file in my repository, but it is not working. 
`
{
    "schedule": "daily",
    "ignoredFiles": [
        "*.jpg",
        "image1.jpg",
        "public/special_images/*",
    ],
    "aggressiveCompression": "true"
    "compressWiki": "true"
}
`

**Comment by Andrewshin-7th-technology-student** [comment]
I have changed it to now

'{
  "minKBReduced": 0
}'

**Comment by JosefHobler** [comment]
I have the same issue probably, did you manage to solve it?

**Comment by Andrewshin-7th-technology-student** [comment]
no, it still is not working. I am requesting again, but it is not updating. Could it be about my branch?

**Comment by Andrewshin-7th-technology-student** [comment]
Here, you can see my repo,
[](https://github.com/Andrewshin-7th-technology-student/build-CI)

**Comment by JosefHobler** [comment]
For me, It was working properly. The only thing I can think of why it stopped working was that I have changed my github username. However I spent like 6 hours trying to resolve it, logging in, logging out, uninstalling, providing .imgbotconfig, etc... Nothing worked.

**Comment by JosefHobler** [comment]
That is probably not issue for me. When Imgbot was working, it gave new pull request everytime I did git push with a new blog article that had DALLE 3 Images. Now it just stopped and it has been already several months without any pull request.

**Comment by Andrewshin-7th-technology-student** [comment]
My repository has not configured with the Imgbot file yet, I think. Do you have any idea why it wont create any pr for my images if I'm using regular git?

**Comment by Andrewshin-7th-technology-student** [comment]
I don't know if this will work.
![image](https://github.com/user-attachments/assets/8bbe8324-4168-4d1d-b5b9-b8ae5588355a)

**Comment by JosefHobler** [comment]
I would probably wait like 3 days and see if it works.

This is my .imgbotconfig (added the minKBReduced today): 
{
  "schedule": "daily",
  "aggressiveCompression": "false",
  "minKBReduced": 0
}

**Comment by Andrewshin-7th-technology-student** [comment]
I think something in my repository is bugging my bots. My dependabot has stopped working to.

**Comment by JosefHobler** [comment]
But imgbot was working without .imgbotconfig file fine. Idk if it will help.

**Comment by JosefHobler** [comment]
Imgbot should have some customer service...

**Comment by Andrewshin-7th-technology-student** [comment]
hm i will try removing .imgbotconfig file to see if it helps or not.

**Comment by Andrewshin-7th-technology-student** [comment]
Mabye this title is needed inorder to open a pr to configure the name?/??????

{
    "prTitle": "My title"
}

**Comment by JosefHobler** [comment]
I have no clue, all I can say that I added .imgbotconfig only as a possible fix to the problem.

Basically it was working without .imgbotconfig, then it suddenly stopped working and so I tried fixing it and one of the fixes was adding .imgbotconfig. However everything was unsuccessful.

**Comment by JosefHobler** [comment]
Idk, lets wait 3 days and see.

**Comment by Andrewshin-7th-technology-student** [comment]
ok

**Comment by Andrewshin-7th-technology-student** [comment]
hey, quick question, does the file size matter to imgbot for configuration and opening pr's? I just put 
{
    "minKBReduced": null
}

**Comment by Andrewshin-7th-technology-student** [comment]
😅

**Comment by Andrewshin-7th-technology-student** [comment]
can i see your repo

**Comment by JosefHobler** [comment]
Its private, lets just wait 3 days

**Comment by Andrewshin-7th-technology-student** [comment]
...ok

**Comment by ChinaGodMan** [comment]
I installed it, and the website also showed that it was scanned, but there was no PR, and reinstalling imgbot was useless. Is this a wrong operation?

**Comment by kcak11** [comment]
Not sure if this bot is actually well-maintained now.

The last commit to master was on `Fri Nov 11 16:46:44 2022 +0200`

@GrigoreMihai can you please provide guidance. Thanks in advance.

**Comment by Andrewshin-7th-technology-student** [comment]
still no pr for me...

**Comment by Andrewshin-7th-technology-student** [comment]
no pr still

**Comment by JosefHobler** [comment]
I got nothing as well... Idk the imgbot is probably broken and not maintained.

**Comment by JosefHobler** [comment]
Are you on student license? Because I am. Maybe its because of that.

**Comment by JosefHobler** [comment]
I activated free trial on imgbot Agency plan. Lets see if I get a PR request.

**Comment by Andrewshin-7th-technology-student** [comment]
im not sure...i think im on...hmmmmm idk  whats wrong maybe your right I think imgbot servers are broken.

**Comment by Andrewshin-7th-technology-student** [comment]
and AI says this:
![image](https://github.com/user-attachments/assets/133e38d0-c1b9-4618-bd86-ecbdc8ed4f06)
So in short summary, licenses doesn't affect imgbots operation...so I think the servers are down. :<

**Comment by Andrewshin-7th-technology-student** [comment]
I have reached out to them again. hopefully they will help. :|

**Comment by JosefHobler** [comment]
Yeah, I am already 3 days on Agency tier free trial and still no pull request. :/

**Comment by JosefHobler** [comment]
I will probably use some alternative to imgbot

**Comment by Andrewshin-7th-technology-student** [comment]
AI said this
![image](https://github.com/user-attachments/assets/763ff35d-39cf-4ec0-a59f-f0baddea5b78)

**Comment by Andrewshin-7th-technology-student** [comment]
never mind these dont exsist

**Comment by Andrewshin-7th-technology-student** [comment]
can you let me know any github apps that are similar?

**Comment by JosefHobler** [comment]
Yeah, I got in on my Todo, when I get to it, I will let you know.

**Comment by GrigoreMihai** [comment]
Hi guys, I'm not currently actively looking over this, could only help you with parts that are related to the open source code. I looked over this caused I got tagged.

Hi @selul, not sure who is currently checking this so I'm tagging you here. Sorry if I shouldn't have tagged you.  

It looks like the job that was rebuilding the compress workers is failing. 

It didn't created PRs for some time: https://github.com/search?o=desc&q=imgbot&s=created&type=Issues .
